thewalrus/tests/test_quantum.py

```diff
@@ -1102,3 +1105,182 @@ def test_fidelity_wrong_shape():
         ValueError, match="The inputs have incompatible shapes"
     ):
         fidelity(mu, cov1, mu, cov2)
+
+
+@pytest.mark.parametrize("hbar", [0.5, 1, 2, 1.6])
+@pytest.mark.parametrize("alpha", np.random.rand(3, 10) + 1j * np.random.rand(3, 10))
+def test_expectation_normal_ordered_coherent(alpha, hbar):
+    """Test the correct evaluation of the normal ordered expectation value for a product of coherent states"""
+    beta = np.concatenate([alpha, np.conj(alpha)])
+    means = Means(beta, hbar=hbar)
+    cov = np.identity(len(beta)) * hbar / 2
+    pattern = np.random.randint(low=0, high=3, size=len(beta))
+    result = normal_ordered_expectation(means, cov, pattern, hbar=hbar)
+    np.allclose(result, np.prod(np.conj(beta) ** pattern))
+
+
+@pytest.mark.parametrize("hbar", [0.5, 1, 2, 1.6])
+@pytest.mark.parametrize("r", np.random.rand(5))
+@pytest.mark.parametrize("phi", 2 * np.pi * np.random.rand(5))
+def test_single_mode_squeezed(r, phi, hbar):
+    """Tests the correct results are obtained for a single mode squeezed state"""
+    S = squeezing(r, phi)
+    cov = 0.5 * hbar * S @ S.T
+    means = np.zeros([2])
+
+    patterns = [
+        [0, 0],
+        [1, 1],
+        [0, 1],
+        [1, 0],
+        [0, 2],
+        [2, 0],
+        [0, 4],
+        [4, 0],
+        [2, 2],
+        [3, 1],
+        [1, 3],
+    ]
+
+    adxa = np.sinh(r) ** 2
+    a = 0
+    a2 = -0.5 * np.exp(1j * phi) * np.sinh(2 * r)
+    a4 = 3 * np.exp(2j * phi) * (0.5 * np.sinh(2 * r)) ** 2
+    ad2xa2 = (np.cosh(r) ** 2 + 2 * np.sinh(r) ** 2) * np.sinh(r) ** 2
+    ad3xa = -3 * np.exp(-1j * phi) * np.cosh(r) * np.sinh(r) ** 3
+
+    expected = [
+        1,
+        adxa,
+        a,
+        np.conj(a),
+        a2,
+        np.conj(a2),
+        a4,
+        np.conj(a4),
+        ad2xa2,
+        ad3xa,
+        np.conj(ad3xa),
+    ]
+
+    for pattern, value in zip(patterns, expected):
+        result = normal_ordered_expectation(means, cov, pattern, hbar=hbar)
+        assert np.allclose(result, value)

```

Comment:
Just a thought on this: by having the for loop, it might not be apparent for a failing test case, which `pattern` and `value` failed exactly. This could be achieved when both `pattern` and `value` were received as a part of parametrization.

This is not easy, however, in this case, as there is a parameter dependency happening (e.g. `adxa` that is contained in `patterns` depends on another parameter `r`.

One way could be defining `hbar`, `r` and `phi` before each function and then parameterizing `pattern` and `value`. Leaving it up for consideration, but as this would mean quite a modification not a dealbreaker for me.